src/Service/Apps/Quap/QuapService.php

```diff
@@ -183,40 +187,242 @@ public function getAnswers(Group $group, ?\DateTimeImmutable $dateTime): Answers
+                continue;
+            }
+
+            $parentGroup = $group->getParentGroup();
+            if ($parentGroup == null) {
+                $parentlessDepartments[] = $quap;
+                continue;
+            }
+            $parentGroupId = $parentGroup->getId();
+
+            $cantonId = $group->getCantonId();
+            if ($cantonId === null) {
+                $parentlessDepartments[] = $quap;
+            }
+
+            $cantonalHierarchy = $nestedDepartmentsMap[$cantonId] ?? null;
+            if ($cantonalHierarchy === null) {
+                $parentlessDepartments[] = $quap;
+                continue;
+            }
+
+            $found = false;
+            foreach ($cantonalHierarchy->getChildren() as $regionalHierarchy) {
+                $regionalQuap = $regionalHierarchy->getParent();
+                if ($regionalQuap->getGroup()->getId() === $parentGroupId) {
+                    $regionalHierarchy->addChild(new HierarchyDTO($quap));
+                    $found = true;
+                    break;
+                }
+            }
+
+            if (!$found) {
+                $parentlessDepartments[$cantonId] = $quap;
             }
-        } elseif ($parentGroupType === GroupType::REGION) {
-            $ids = $this->statisticGroupRepository->findAllRelevantChildGroups($group->getId(), [GroupType::DEPARTMENT]);
-        } else {
-            throw new \Exception();
         }
 
-        $answers = $this->quapRepository->findAllAnswers($ids, $date !== null ? $date->format('Y-m-d') : null);
+        $nestedQuaps = array_values($nestedDepartmentsMap);
+        if (count($parentlessDepartments) > 0) {
+            $nestedQuaps[] = new HierarchyDTO(null, $this->mapToRegionalHierarchyAggregated($parentlessDepartments));
+        }
+
+        return $nestedQuaps;
+    }
+
+
+    /**
+     * @param AggregatedQuap[] $quaps
+     * @return HierarchyDTO<AggregatedQuap>[]
+     */
+    private function mapToRegionalHierarchyAggregated(array $quaps): array
+    {
+        /**
+         * maps a group_id of a region to a HierarchyDTO of AggregatedQuap
+         * @var array<int, HierarchyDTO<AggregatedQuap>> $nestedDepartmentsMap
+         */
+        $nestedDepartmentsMap = array();
+        /**
+         * @var AggregatedQuap[] $parentlessDepartments
+         */
+        $parentlessDepartments = array();
+
+        foreach ($quaps as $quap) {
+            $group = $quap->getGroup();
+            if ($group->getGroupType()->getGroupType() === GroupType::REGION) {
+                $nested = $nestedDepartmentsMap[$group->getId()] ?? null;
+                if ($nested === null) {
+                    $nestedDepartmentsMap[$group->getId()] = new HierarchyDTO($quap);
+                    continue;
+                }
+                $nested->setParent($quap);
+                continue;
+            }
+
+            $parentGroup = $group->getParentGroup();
+            if ($parentGroup == null) {
```

Comment:
this check misses a `=`. please also quickly check if there are other `==` checks that didn't catch my attention